go/merkletree2/config.go

```diff
@@ -20,27 +41,45 @@ type Config struct {
 	// bitsPerIndex divides keyByteLength*8
 	keysByteLength int
 
-	// Construct a new object to unmarshal values into
+	// valueConstructor is an interface to construct empty values to be used for
+	// deserialization.
 	valueConstructor ValueConstructor
 }
 
-// NewConfig makes a new config object. It takes a a Hasher (for example
-// sha512.Sum512), logChildrenPerNode which is the base 2 logarithm of the
-// number of children per interior node, valuesPerLeaf the maximum number of
-// entries in a leaf before the leaf is split into multiple nodes (at a lower
-// level in the tree), keyByteLength the length of the Keys which the tree
-// will store, and a valueConstructor (so that typed values can be pulled out
-// of the Merkle Tree).
-func NewConfig(h Hasher, logChildrenPerNode uint8, valuesPerLeaf int, keysByteLength int, valueConstructor ValueConstructor) (Config, error) {
+// NewConfig makes a new config object. It takes a a Hasher, logChildrenPerNode
+// which is the base 2 logarithm of the number of children per interior node,
+// maxValuesPerLeaf the maximum number of entries in a leaf before the leaf is
+// split into multiple nodes (at a lower level in the tree), keyByteLength the
+// length of the Keys which the tree will store, and a valueConstructor (so that
+// typed values can be pulled out of the Merkle Tree).
+func NewConfig(h Encoder, useBlindedValueHashes bool, logChildrenPerNode uint8, maxValuesPerLeaf int, keysByteLength int) (Config, error) {
```

Comment:
Want to assert that `logChildrenPerNode > 0` since we divide by `numChildren-1` in a lot of places, which would crash.